### PR TITLE
Add NewtonDynamics

### PR DESCRIPTION
**Pull request by SirLynix**

### CODE REVIEW

**Comment by SirLynix** [comment]
I'm beginning to think it would be easier to make a xmake.lua file to compile it.

File: packages/n/newtondynamics/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    end)
+
+    on_install("windows", "linux", "macosx", "mingw", "android", function (package)
+        -- prevent cmakelists.txt from overwriting CMAKE_INSTALL_PREFIX
+        io.replace("CMakeLists.txt", [[set(CMAKE_INSTALL_PREFIX "win64sdk" CACHE PATH "..." FORCE)]], "", {plain = true})
+        io.replace("CMakeLists.txt", [[set(CMAKE_INSTALL_PREFIX "win32sdk" CACHE PATH "..." FORCE)]], "", {plain = true})
+
+        local configs = {"-DNEWTON_BUILD_SANDBOX_DEMOS=OFF", "-DNEWTON_BUILD_CORE_ONLY=ON"}
+        table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
+        table.insert(configs, "-DDNEWTON_BUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
```
Comment (waruqi) [inline packages/n/newtondynamics/xmake.lua:51]: `-DNEWTON_BUILD_SHARED_LIBS `, not `-DD`

**Comment by waruqi** [comment]
> libdgCore.so: undefined reference to `pthread_create'
/usr/bin/ld: /home/ruki/.xmake/packages/n/newtondynamics/v3.14c/6a5e3ff14e21470782bab40bfd1b098d/lib/libdgPhysics.so: undefined reference to `dlopen'

if is shared then
    Maybe we should patch CMakelist.txt to add pthread link to -ldgCore target and add -ldl to dgPhysics target.
end

**Comment by waruqi** [comment]
> I'm beginning to think it would be easier to make a xmake.lua file to compile it.

maybe, you can try it.

File: packages/n/newtondynamics/xmake.lua (hunk 1 above)
Comment (SirLynix) [reply packages/n/newtondynamics/xmake.lua:51]: 😅

File: packages/n/newtondynamics/port/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        end
+        if is_plat("android") then
+            add_defines("_ARM_VER")
+        end
+    elseif is_plat("macosx", "iphoneos") then
+        add_defines("_MACOSX_VER")
+    end
+
+    if is_plat("windows") then
+        if get_config("kind") == "static" then
```
Comment (waruqi) [inline packages/n/newtondynamics/port/xmake.lua:35]: is_kind("static")

Comment (SirLynix) [reply packages/n/newtondynamics/port/xmake.lua:35]: Fixed.
Also, is it normal that this define, while being public, is not included in the package info? I had to use on_load in the package to have it.

Comment (waruqi) [reply packages/n/newtondynamics/port/xmake.lua:35]: If you need to export definitions to users, you should set it in the external package scope through add_defines or set it in on_load.

The definition in port/xmake.lua only affects the compilation of the package itself, and does not export the definition to the user project.

**Comment by waruqi** [comment]
android not add pthread links

**Pull request merged**